Sources/ClientRuntime/Networking/Http/URLSession/FoundationStreamBridge.swift

```diff
@@ -44,80 +50,99 @@ class FoundationStreamBridge: NSObject, StreamDelegate {
     /// A Logger for logging events.
     private let logger: LogAgent
 
-    /// Actor used to ensure writes are performed in series.
-    actor WriteCoordinator {
+    /// Actor used to ensure writes are performed in series, one at a time.
+    private actor WriteCoordinator {
         var task: Task<Void, Error>?
 
-        /// `true` if the readable stream has been found to be empty, `false` otherwise.  Will flip to `true` if the readable stream is read,
-        /// and `nil` is returned.
-        var readableStreamIsEmpty = false
-
-        /// Sets stream status to indicate the stream is empty.
-        func setReadableStreamIsEmpty() async {
-            readableStreamIsEmpty = true
-        }
```

Comment:
This flag moved to the `FoundationStreamBridge` because having it on this actor is redundant - the WriteCoordinator's `perform()` method enforces exclusive access so having this flag on the Actor is just superfluous.